torch/_inductor/codegen/common.py

```diff
@@ -52,14 +52,18 @@ def _print_Pow(self, expr):
         # Pow() confuses triton
         base, exp = expr.args
         base = self._print(base)
-        assert exp.is_integer
-        exp = int(exp)
-        if exp > 0:
-            return "*".join([self.paren(base)] * exp)
-        elif exp < 0:
-            return "1/" + self.paren("*".join([self.paren(base)] * abs(exp)))
-        else:  # exp == 0
-            return "1"
+        if exp.is_integer:
+            exp = int(exp)
+            if exp > 0:
+                return "*".join([self.paren(base)] * exp)
+            elif exp < 0:
+                return "1/" + self.paren("*".join([self.paren(base)] * abs(exp)))
+            else:  # exp == 0
+                return "1"
+        elif isinstance(exp, sympy.core.numbers.Half):
+            return f"tl.libdevice.sqrt({self.paren(base)}.to(tl.float32))"
+        else:
+            return f"tl.libdevice.pow({base}, {self._print(exp)})"
```

Comment:
I'm not sure about this. The changes here are consistent with @nkaretnikov's changes in https://github.com/pytorch/pytorch/pull/95698 and https://github.com/pytorch/pytorch/pull/95773 but honestly this class of changes gives me the jeebies.

One thing that particularly annoys me is the conversion of base into a float32 before sqrt'ing it. This feels... bad. Why not a float64? Well, that actually doesn't work because downstream seems to expect this to be a float32. How do I even tell what the width of the type is supposed to be here. Sympy is just not giving us enough rope.

Reply:
This is `ExprPrinter`, `tl.libdevice.*` should be in `TritonPrinter`, no?